stagemonitor-core/src/main/java/org/stagemonitor/core/configuration/SpringCloudConfigConfigurationSource.java

```diff
@@ -0,0 +1,206 @@
+		this(httpClient, configServerAddress, applicationName, DEFAULT_PROFILE);
+	}
+
+
+	/**
+	 * Creates an ConfigServerConfigurationSource instance and fetches the configuration initially via a http GET
+	 *
+	 * @param httpClient          Instance of HttpClient
+	 * @param configServerAddress Http or https address of the config server
+	 * @param applicationName     Name of the application to fetch config for
+	 * @param profile             Name of profile to be used. If empty or null "default" will be used
+	 */
+	public SpringCloudConfigConfigurationSource(HttpClient httpClient, String configServerAddress, String applicationName, String profile) {
+
+		if (StringUtils.isEmpty(configServerAddress) || !configServerAddress.startsWith("http")) {
+			throw new IllegalArgumentException("Invalid config server address: " + configServerAddress);
+		}
+
+		if (StringUtils.isEmpty(applicationName)) {
+			throw new IllegalArgumentException("The application name must not be empty.");
+		}
+
+		this.httpClient = httpClient;
+		this.properties = new HashMap<String, String>();
+		this.profile = profile;
+		this.applicationName = applicationName;
+
+		this.configAddress = getFullQualifiedConfigUrl(configServerAddress, applicationName, profile);
+		reload();
+	}
+
+
+	/**
+	 * Return the property value for a give key
+	 *
+	 * @param key the property key
+	 * @return value of property, or null
+	 */
+	@Override
+	public String getValue(String key) {
+		return properties.get(key);
+	}
+
+
+	/**
+	 * Returns the config address of this config source
+	 *
+	 * @return full config address including service and profile
+	 */
+	@Override
+	public String getName() {
+		return configAddress;
+	}
+
+
+	public String getProfile() {
+		return profile;
+	}
+
+
+	public String getApplicationName() {
+		return applicationName;
+	}
+
+
+	@Override
+	public void reload() {
+
+		logger.debug("Requesting configuration from: " + configAddress);
+
+		Map<String, String> newProperties = httpClient.send("GET", configAddress, new HashMap<String, String>(), null, new HttpClient.ResponseHandler<Map<String, String>>() {
+			@Override
+			public Map<String, String> handleResponse(HttpRequest<?> httpRequest, InputStream is, Integer statusCode, IOException e) throws IOException {
+				if (e != null || statusCode != 200 || is == null) {
+					logger.warn("Couldn't GET configuration. " + configAddress + " returned " + statusCode);
+					return null;
+				}
+
+				return getConfigurationFromInputStream(is);
+			}
+		});
+
+		// Only updating the config, if there's valid server response
+		if (newProperties != null) {
+			logger.debug("reloading ConfigServerConfigurationSource with: " + newProperties);
+			properties.clear();
+			properties.putAll(newProperties);
+		}
+	}
+
+
+	/**
+	 * Processes an input stream and converts the incoming string to a Map<String, String>
+	 */
+	private Map<String, String> getConfigurationFromInputStream(InputStream is) {
```

Comment:
Use `java.util.Properties#load(java.io.InputStream)` instead of parsing the key/value pairs yourself. For example, what about multi-line configurations?

Reply:
Didn't know of this. Makes the code a lot simpler! thx